nav2_core/CMakeLists.txt

```diff
@@ -2,36 +2,47 @@ cmake_minimum_required(VERSION 3.5)
 project(nav2_core)
 
 find_package(ament_cmake REQUIRED)
-find_package(nav2_common REQUIRED)
-find_package(rclcpp REQUIRED)
-find_package(rclcpp_lifecycle REQUIRED)
-find_package(std_msgs REQUIRED)
 find_package(geometry_msgs REQUIRED)
+find_package(nav2_behavior_tree REQUIRED)
+find_package(nav2_common REQUIRED)
 find_package(nav2_costmap_2d REQUIRED)
-find_package(pluginlib REQUIRED)
-find_package(tf2_ros REQUIRED)
 find_package(nav2_util REQUIRED)
 find_package(nav_msgs REQUIRED)
-find_package(nav2_behavior_tree REQUIRED)
+find_package(pluginlib REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(rclcpp_lifecycle REQUIRED)
+find_package(tf2_ros REQUIRED)
 
 nav2_package()
 
-set(dependencies
-  rclcpp
-  rclcpp_lifecycle
-  std_msgs
-  geometry_msgs
-  nav2_costmap_2d
-  pluginlib
-  visualization_msgs
-  nav_msgs
-  tf2_ros
-  nav2_util
-  nav2_behavior_tree
+add_library(nav2_core INTERFACE)
+target_include_directories(nav2_core
+  INTERFACE
+    "$<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>"
+    "$<INSTALL_INTERFACE:include/${PROJECT_NAME}>"
+)
+target_link_libraries(nav2_core INTERFACE
+  ${geometry_msgs_TARGETS}
+  nav2_behavior_tree::nav2_behavior_tree
```

Comment:
Nav2's side of things uses the old CMake rather than the newer in the main branch https://github.com/ros-navigation/navigation2/blob/jazzy/nav2_behavior_tree/CMakeLists.txt. Will this actually work considering that?